### PR TITLE
azure: prioritize location with quota availability

### PR DESCRIPTION
**Pull request by pupacha**
Current code returns the awaitable_candidates of the first location from the allowed_locations list even if other locations have available_candidates. This change will iterate through all the allowed_locations and returns a matching available_candidate. If no matching is found, it iterates through locations again and gives an awaitable_candidate.

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -521,34 +521,34 @@ def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
         all_awaitable: bool = False
         errors: List[str] = []
 
-        for location in allowed_locations:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py:524]: This loop should check all locations already. Why move the loop into `_get_azure_capabilities`

**Comment by pupacha** [comment]
@squirrelsc I noticed LISA doesn't use other locations even if there is available quota, when multiple locations are passed. With these changes, LISA would prefer a location with available quota.

File: lisa/sut_orchestrator/azure/platform_.py (hunk 1 above)
Comment (pupacha) [reply lisa/sut_orchestrator/azure/platform_.py:524]: The loop was in `_prepare_environment` method. and `_get_azure_capabilities` method was returning an `awaitable_candidate` (if applicable) for the first location of the list. And even if the other locations have enough available quota, it was using the awaitable_candidate.

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py:524]: The L538 `if all(isinstance(x, schema.NodeSpace) for x in caps):` is to check if all are available resources. If it's not, the loop takes the next location. Please check why it doesn't work as expected.

Comment (pupacha) [reply lisa/sut_orchestrator/azure/platform_.py:524]: IIUC, the check is to verify the object type. How does L538 verify if the quota is available or not? 
This https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/platform_.py#L2592 could possibly return awaitable result.

**Comment by pupacha** [comment]
https://github.com/microsoft/lisa/pull/3174 fixes the same issue mentioned in this PR. Closing PR.

**Pull request closed without merging**